pkg/util/validation/validation.go

```diff
@@ -21,6 +21,9 @@ const LabelValueMaxLength int = 63
 // ValidatePropagationSpec validates a PropagationSpec before creation or update.
 func ValidatePropagationSpec(spec policyv1alpha1.PropagationSpec) field.ErrorList {
 	var allErrs field.ErrorList
+	if spec.Preemption == policyv1alpha1.PreemptAlways {
+		allErrs = append(allErrs, ValidateResourceSelectors(spec.ResourceSelectors, field.NewPath("spec").Child("resourceSelectors"))...)
+	}
```

Comment:
I'd like to validate the most common case first, I mean put this logic to the end of this function.